🐛 [HOTFIX] #78: 커뮤니티 충돌 문제 수정

kimeunsom opened the pull request:
## #⃣ 연관된 이슈

> ex) #78, #107

## 📝 작업 내용

삭제된 쿼리, 중복 쿼리 및 오타 정리

### 📸 스크린샷 (선택)

## 💬 리뷰 요구사항(선택)
이제 모임 api들이 모두 정상 작동 해야합니다 !

- 참여 인원 넘쳤을 때, 모임 가입 제한
- 생성 location 타입 유효성 검증 후에 모임 생성
- 약속 설정 body 유효성 검증해서 받기. (좌표, 주소, 날짜)
- 모임 참여여부 불린값 조회 쿼리 수정

kimeunsom committed:
FIX : 참여 충돌 문제 수정
- 삭제된 dao 복구 및 오류 수정

kimeunsom committed:
refactor : 삭제 및 탈퇴 오류 수정

kimeunsom committed:
refactor: 채팅방 상세, 커뮤니티 약속 설정 수정
- 오타 정정

ajwoong reviewed (changes requested):
말씀드린 부분만 고쳐주시면 될것같습니다!

kimeunsom committed:
bug : 현재 참여인원 카운트 로직 에러 수정
- 탈퇴한 유저는 제외하고 카운트하도록 GET_COMMUNITY_DETAILS 에 where절 수정

kimeunsom committed:
refactor : 입력값 유효성 검사 추가
- 약속 설정 날짜, 좌표값이 올바른 값이 입력되는지 유효성 검사 후 약속 설정
- 모임 생성 시에 location enum값 검증
- + 탈퇴자  참여 인원으로 산정되던  오류 수정

kimeunsom committed:
refactor : 인원수 초과 에러 이전에 모임 참여 여부 에러처리부터 진행

kimeunsom committed:
refactor: 모임 날짜 설정 시, iso 8601 표준 값으로 받도록 설정

Sunho12 reviewed (changes requested):
- communities.service.js 파일에서
getCommunityDetailsService 155번째 줄
`isUserParticipating = await checkUserInCommunity(communityId, userId);
    }`
dao. 추가해주세요 !



- 방장임에도 불구하고 isParticipating 이 0으로 뜹니다! - 커뮤니티상세조회
<img width="528" alt="image" src="https://github.com/user-attachments/assets/01c34fac-1f9d-4dc6-a98a-09628647b846">

kimeunsom committed:
refactor : 오류 수정
- 표준시 설정 정규표현식 밀리초까지 받아오도록 수정
- 모임 참여여부 쿼리 수정. treu, false 올바르게 받아와짐

Sunho12 reviewed (approved):
말씀드린 부분 수정 확인했습니다 ! 수고하셨어요 👍

ajwoong reviewed (approved):
말씀드린 부분 수정된거 확인했습니다! 고생하셨어요~~
